runtime/src/main/java/io/quarkiverse/freemarker/runtime/FreemarkerBuildConfig.java

```diff
@@ -15,16 +22,140 @@
      * Comma-separated list of absolute resource paths to scan recursively for templates.
      * All tree folder from 'resource-paths' will be added as a resource.
      * Unprefixed locations or locations starting with classpath will be processed in the same way.
+     * <p>
+     * Defaults relevant for this option are documented on {@code quarkus.freemarker.base-path}
+     *
+     * @deprecated Use template set group of options instead ({@code quarkus.freemarker.<template-set-name>.base-path}
+     *             {@code quarkus.freemarker.<template-set-name>.includes} and
+     *             {@code quarkus.freemarker.<template-set-name>.excludes})
      */
-    @ConfigItem(defaultValue = "freemarker/templates")
-    public List<String> resourcePaths;
+    @ConfigItem
+    public Optional<List<String>> resourcePaths;
 
     /**
      * List of directives to register with format name=classname
-     * 
+     *
      * @see freemarker.template.Configuration#setSharedVariable(String, TemplateModel)
      */
     @ConfigItem
     public Map<String, String> directive;
 
+    /**
+     * The default template set
+     */
+    @ConfigItem(name = ConfigItem.PARENT)
+    public TemplateSet defaultTemplateSet;
+
+    /**
+     * Additional named template sets.
+     */
+    @ConfigDocSection
+    @ConfigDocMapKey("template-set-name")
+    @ConfigItem(name = ConfigItem.PARENT)
+    public Map<String, TemplateSet> namedTemplateSets;
+
+    @ConfigGroup
+    public static class TemplateSet {
+        /**
+         * The base path of this template set.
+         * <p>
+         * Use slash ({@code /}) as a path separator on all platforms. The value must not start with a slash.
+         * <p>
+         * {@code includes} and {@code excludes} are relative to {@code base-path}.
+         * <p>
+         * Do not set any {@code base-path} value if you want {@code includes} and {@code excludes} to be relative to
+         * root resource path.
+         * <p>
+         * <strong>Defaults</strong>
+         * <p>
+         * The following defaults are effective only of <strong>none</strong> of {@code quarkus.freemarker.base-path},
+         * {@code quarkus.freemarker.includes}, {@code quarkus.freemarker.excludes} and
+         * {@code quarkus.freemarker.resource-paths} is set. This is to achieve the backwards compatibility with the
+         * behavior of {@code quarkus.freemarker.resource-paths} before Quarkus Freemarker 0.2.0.
+         * <table>
+         * <tr>
+         * <th>Option</th>
+         * <th>Default value</th>
+         * </tr>
+         * <tr>
+         * <td>{@code quarkus.freemarker.base-path}</td>
+         * <td>{@code freemarker/templates}</td>
+         * </tr>
+         * <tr>
+         * <td>{@code quarkus.freemarker.includes}</td>
+         * <td>{@code **}</td>
```

Comment:
I think we need to replace `**` by something like `<required>`.

Reply:
The table informs about default values in case none of base-path, includes, excludes and resource-paths is set. For that case, the default of includes is indeed `**`. I admit the table is easy to misinterpret, so I have added another column to explain defaults that are effective otherwise.